src/test/java/pl/marcin/project/repository/CupRepositoryListBasedTest.java

```diff
@@ -0,0 +1,228 @@
+package pl.marcin.project.repository;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import pl.marcin.project.model.Cup;
+import pl.marcin.project.model.Customer;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CupRepositoryListBasedTest {
+    private CupRepository cupRepository = new CupRepositoryListBased();
```

Comment:
I encourage you to extract some common `ids`, `cups` to fields or initialize them in` BeforeEach` method. When they are initialized from scratch in every test, they take up space. That also indicates that specific `given` conditions for all tests are always distinct and specific - but they are not, many tests reuse the same `id` and their cups consist of the same data. This tells us we can extract them to fields to minimalise code.

Same applies to `CustomerRepositoryListBasedTest.java`

Reply:
ok